### ISSUE TITLE
New to Go, Not working on Ubuntu 18

### ISSUE THREAD

**Issue by alteredorange** [MontFerret/ferret#32]
I'm new to this go stuff.  I tried installing this on Ubuntu 18.  First installing go, and then trying to make ferret... Would it be possible to post a completely newbie guide with all the command line steps? Would be much appreciated, thanks!

**Comment by ziflex** [comment]
Hey, I would be happy to help you.
What error are you getting?
Have you managed to install Go, Antlr4 and Make?

**Comment by alteredorange** [comment]
I don't think I installed Antlr4...  I'm using the one-click docker install on DigitalOcean.  Make is pre-installed. There's a bunch of different tutorials showing different ways to install go and dep.  I was getting path errors (ferret wasn't in go path), then dep errors, and then make errors.  Installed go again and got more path errors :/

**Comment by krishnakarthik1309** [comment]
Hi, I just started looking into this project today and I was able to set it up in my ubuntu by following these steps.
1.You need to setup `GOPATH` environment variable. Refer [this](https://github.com/golang/go/wiki/SettingGOPATH)
2.You need to clone this repository into `$GOPATH/src/github.com/MontFerret/ferret` and `cd` to this directory.
3. Install `dep` using `curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh`
4. Install `antlr4` by following this [link](https://www.techinfected.net/2016/03/how-to-install-antlr4-in-ubuntulinux.html).

Run `make`

**Comment by alteredorange** [comment]
I'm getting a dep error doing this.  It looks like it's installing dep in /root/go/bin instead of usr/local/go/bin, but I've added gopath to /etc/profile, ~/.profile, ~/.bash_profile, but it still goes to root, and when I use the "make build" command, I get: 
`dep ensure
make: dep: Command not found
Makefile:20: recipe for target 'install' failed
make: *** [install] Error 127`

**Comment by krishnakarthik1309** [comment]
As you are using ubuntu adding gopath in `~/.bashrc` or `~/.profile` should be enough.
Also add this `export PATH=$PATH:$GOPATH/bin`. Make sure that you create bin folder using `mkdir $GOPATH/bin` and run `source ~/.bashrc` or `source ~/.profile`.  Then continue from 3rd step(installing dep).

These steps will install dep into `$GOPATH/bin`, as we added this folder into `$PATH` variable this time `dep` command will be executed.

**Comment by alteredorange** [comment]
okay adding `export GOPATH=/usr/local/go` and `export PATH=$PATH:$GOPATH/bin` to `~/.bashrc` fixed the dep error, but now I get these errors:
`dep ensure
go vet ./cmd/... ./pkg/...
warning: GOPATH set to GOROOT (/usr/local/go) has no effect
# runtime/cgo
exec: "gcc": executable file not found in $PATH
Makefile:43: recipe for target 'vet' failed
make: *** [vet] Error 2`

**Comment by alteredorange** [comment]
Oh wait, I fixed that error with `sudo apt-get install gcc` but now it's failing on Antlr... 
`dep ensure
go vet ./cmd/... ./pkg/...
warning: GOPATH set to GOROOT (/usr/local/go) has no effect
go generate ./pkg/...
warning: GOPATH set to GOROOT (/usr/local/go) has no effect
pkg/parser/parser.go:1: running "antlr4": exec: "antlr4": executable file not found in $PATH
Makefile:26: recipe for target 'generate' failed
make: *** [generate] Error 1`

**Comment by krishnakarthik1309** [comment]
I am not sure about that gcc error, but `GOPATH` is not the path where `go` executable is there, it can be any folder. I suggest you to set it to `$HOME/go`. `GOPATH/src` is where all your go should exist. After doing this follow the above steps.

**Comment by krishnakarthik1309** [comment]
Ok.. I am also looking into that `antlr4` error, previously I have written a 1 line shell script to make antlr4 as executable. I will comment when I find the solution.

**Comment by krishnakarthik1309** [comment]
For some reason adding antl4 as alias is not working. I have a work around for antlr4.

**Comment by alteredorange** [comment]
Yeah, this isn't an easy setup! Maybe if it were a docker image or something...

**Comment by krishnakarthik1309** [comment]
@alteredorange Hey I am not sure there might be simpler method, I just started out today with this one. @ziflex may be able to provide more insights. If it is going to take all these steps then probably one can write a small shell script for installing all these.

**Comment by ziflex** [comment]
@alteredorange @krishnakarthik1309 hey guys, you do not really need to generate parser, since it's a part of the repo. You just need to install dependencies (``make install``) and compile the project (``make compile``).

I will update README soon.

**Comment by alteredorange** [comment]
Wow that worked! (`make install` and `make compile`).  You also need to update the readme to `go run ./cmd/main.go` instead of `go run ./cmd/cli/main.go` (at least that's were my main.go got installed).  Now to actually play around with it a bit!  Thanks!

**Comment by ziflex** [comment]
@alteredorange good catch! Yeah, I wrote those snippets before I moved a server into a separate repo.
Glad that you can finally run it!

**Comment by alteredorange** [comment]
Putting this here for future reference, pretty sure these commands will get everything working on an Ubuntu 18 Digital Ocean Droplet with Docker image one click install:
`sudo apt-get update
sudo apt-get -y upgrade
sudo apt-get install gcc
sudo curl -O https://dl.google.com/go/go1.11.1.linux-amd64.tar.gz
sudo tar -xvf go1.11.1.linux-amd64.tar.gz
sudo mv go /usr/local
cd /usr/local/go
GOPATH=/usr/local/go
sudo nano ~/.profile
(add to bottom of profile):
export GOPATH=$HOME/work
export PATH=$PATH:/usr/local/go/bin:$GOPATH/bin
ctrl+x to save and close file
source ~/.profile
mkdir $HOME/work/github.com/MontFerret
cd $HOME/work/github.com/MontFerret
git clone https://github.com/MontFerret/ferret
cd ferret
curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh

sudo apt-get install make
make install
make compile`

**Comment by ziflex** [comment]
@alteredorange can I close the issue?
I refactored the project a bit, now you can install the binary with ``go get github.com/MontFerret/ferret`` command.
